pkg/crc/preflight/preflight_checks_nonlinux.go

```diff
@@ -70,3 +73,13 @@ func olderDaemonVersionRunning() error {
 	}
 	return daemonclient.CheckIfOlderVersion(v)
 }
+
+func checkIfDaemonIsServing() error {
+	return crcerrors.Retry(context.Background(), 15*time.Second, func() error {
+		_, err := daemonclient.GetVersionFromDaemonAPI()
```

Comment:
This can reuse `daemonRunning` from #3151.
And I forgot to mention, but I'm not sure it's worth introducing a `serving` state, so far we've always used `running` when checking if the daemon replies to http requests.

Reply:
yes I can reuse the #3151 but as of that that piece is only in `_darwin` preflight check file so either I have to wait it to merge and then relocate this in `_nonlinux` file.

Reply:
https://github.com/code-ready/crc/pull/3150#discussion_r858899083 same here

Reply:
@cfergeau this shouldn't be block the PR atm (it's improvement to reuse the code and we can do it later also).